client/ayon_core/pipeline/load/plugins.py

```diff
@@ -1,21 +1,26 @@
-import os
+"""Plugins for loading representations and products into host applications."""
+from __future__ import annotations
+
 import logging
+import os
 
-from ayon_core.settings import get_project_settings
 from ayon_core.pipeline.plugin_discover import (
+    deregister_plugin,
+    deregister_plugin_path,
     discover,
     register_plugin,
     register_plugin_path,
-    deregister_plugin,
-    deregister_plugin_path
 )
+from ayon_core.settings import get_project_settings
+
 from .utils import get_representation_path_from_context
 
 
 class LoaderPlugin(list):
     """Load representation into host application"""
 
-    product_types = set()
+    product_types: set[str] = set()
+    product_base_types: set[str] = set()
```

Comment:
I really think this should be `None` so we have way how to find out if `product_base_types` should be used for filtering or  `product_types` should be used for filtering. We should NOT use both.